torch/csrc/distributed/c10d/DispatchOps.cpp

```diff
@@ -0,0 +1,38 @@
+#include <torch/csrc/distributed/c10d/DispatchOps.h>
+
+#include <aten/src/ATen/core/PythonModeTLS.h>
+#include <torch/library.h>
+
+namespace c10d {
+namespace {
+c10::intrusive_ptr<ProcessGroup::Work> broadcast(const c10::intrusive_ptr<ProcessGroup>& process_group,
+    at::TensorList tensors, int64_t root_rank = 0, int64_t root_tensor = 0, int64_t timeout = -1) {
+  auto tensor_vec = tensors.vec();
+  return process_group->broadcast(tensor_vec,
+      BroadcastOptions {root_rank, root_tensor, std::chrono::milliseconds(timeout)});
+}
+
+TORCH_LIBRARY(c10d, m) {
+  // The following ProcessGroup and Work definations are more like declarations.
+  // They don't expose the details of the two classes into TorchScript.
+  m.class_<ProcessGroup>("ProcessGroup")
```

Comment:
Not sure why we need to define `ProcessGroup` and `Work` Python types here. We already have them defined in `init.cpp`. Am I missing something?

Reply:
This is for TorchScript not pybind11.